src/Plugin/MediaEmbedDialog/Image.php

```diff
@@ -327,10 +329,16 @@ public function submitDialogForm(array &$form, FormStateInterface $form_state) {
       'data-entity-embed-display-settings',
     ]);
     $settings = array_filter($settings);
-    // Clean up the display settings, but we still want at least an empty alt
-    // text. This also helps prevent an empty array which converts to an empty
-    // string. An empty string breaks the render portion.
-    $settings['alt_text'] = isset($settings['alt_text']) ? $settings['alt_text'] : '';
+    // Add a simple placeholder. This just prevents the settings from being an
+    // empty string. An empty string causes php notices when displaying.
+    if (!isset($settings['image_style']) && empty($settings['linkit']['href'])) {
+      $settings['p'] = 1;
```

Comment:
What is `p`?

Reply:
Short for placeholder?

Reply:
ya. i was gonna use the word `placeholder` but the smaller, the less storage in the long run.

Reply:
Can you not just set the variable to an empty array instead of a string?

Reply:
interestingly the `$settings` is actually populated with empty values from `linkit` integration. but during save it filters out empty values and then the resulting html markup is `data-entity-embed-display-settings=""` instead of `data-entity-embed-display-settings="{}"`

Reply:
Perhaps make a note of that and change the key to something more searchable. By searchable I mean grepping your codebase. This seems a bit of a bandaid for an issue in linkit and at somepoint, they might fix it. It would be nice to the next developer who is looking for where that key gets set to find it a bit easier.

Reply:
oh its not a linkit issue. i was just proding that as an example that the `$settings` is a populated array already. but yeah i see what you mean by grepping for it. `placeholder` it is.